scripts/install-build-tools.sh

```diff
@@ -14,35 +15,208 @@ if (( $EUID != 0 )); then
     SUDO='sudo'
 fi
 
+# Supporting these versions for buildflow
+PYTHON_VERSIONS=("3.12" "3.11" "3.10")
+ENV_NAME=".py_venv"
+PY_INSTALLED=''
+echo "Python3 versions supported: ${PYTHON_VERSIONS[@]}"
+
+# make a virtual environement to install python packages
+create_venv_install_python() {
+  PY_LOC=$1
+  if [ -z "$PY_LOC" ]; then
+    echo "Python path not provided"
+    exit 1
+  fi
+  PY_VERSION=$2
+  if [ -z "$PY_VERSION" ]; then
+    echo "python version not provided"
+    exit 1
+  fi
+  # remove existing virtual environment if it exists - global var
+  ROOT="$(cd "$(dirname "$0")"/.. && pwd)"
+  ENV_LOCATION="${ROOT}/${ENV_NAME}"
+  if [ -d "${ENV_LOCATION}" ]; then
+    if rm -rf -- "${ENV_LOCATION}"; then
+      echo "Removed existing virtual environment"
+    else
+      echo "Failed to remove existing virtual environment"; exit 1
+    fi
+  fi
+  # install pip for linux
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    if ! $SUDO apt install -y python3-pip; then
+      echo "Failed to install python3-pip"
+      wget https://bootstrap.pypa.io/get-pip.py
+      $SUDO python${PY_VERSION} get-pip.py
+      rm get-pip.py
+    fi
+    # install python3 venv module for linux
+    if ! $SUDO apt install -y "python${PY_VERSION}-venv"; then
+      echo "Failed to install python${PY_VERSION}-venv"
+    else
+      echo "Installed python${PY_VERSION}-venv"
+    fi
+  fi
+  # create virtual environment with specific python version
+  if "${PY_LOC}" -m venv "${ENV_NAME}"; then
+    echo "Virtual environment '${ENV_NAME}' created"
+  else
+    echo "Virtual environment creation failed"; exit 1
+  fi
+  # activate virtual environment
+  source "${ROOT}/scripts/activate-venv.sh"
+  # install python packages
+  if "${ROOT}/${ENV_NAME}/bin/pip" install -r requirements.txt; then
+    echo "Success installing python packages"
+  else
+    echo "Failure installing python packages"
+    deactivate
+    exit 1
+  fi
+  deactivate
+}
+
+
+echo "OS Type: $OSTYPE"
+# macOS install with homebrew
 if [[ "$OSTYPE" == "darwin"* ]]; then
+
+  # macOS does not support running shell scripts as root with homebrew
+  if [ $EUID -eq 0 ]; then
+    echo -e "Mac users should run this script without 'sudo'. Exiting..."
```

Comment:
Is this a danger thing? Or is this just because brew will default fail if run as root